File: core/src/com/questmark/entity/components/SpeedComponent.java
package com.questmark.entity.components;

import com.badlogic.ashley.core.Component;

/**
 * A {@link Component} that represents how fast an entity can move. In other words,
 * the magnitude of the entity's velocity.
 *
 * @author Ming Li
 */
public class SpeedComponent implements Component {

    public float speed;

    public SpeedComponent(float speed) throws IllegalArgumentException {
        if (speed < 0) {
            throw new IllegalArgumentException("Speed cannot be negative");
        }
        this.speed = speed;
    }

}

File: core/src/com/questmark/entity/components/SpeedComponent.kt
package com.questmark.entity.components

import com.badlogic.ashley.core.Component

/**
 * A [Component] that represents how fast an entity can move. In other words,
 * the magnitude of the entity's velocity.
 *
 * @author Ming Li
 */
class SpeedComponent @Throws(IllegalArgumentException::class)
constructor(var speed: Float) : Component {

    init {
        if (speed < 0) {
            throw IllegalArgumentException("Speed cannot be negative")
        }
    }

}
